src/product/index.ts

```diff
@@ -0,0 +1,231 @@
+import api from '../../api'
+import { Product, Response } from '../../types'
+
+async function getProductDetails (productId: string): Promise<Product | Response> {
+  const query = {
+    "json": {
+      "params": {
+        "group": true,
+        "group.field": `groupId`,
+        "group.limit": 10000,
+        "group.ngroups": true,
+        "rows":10
+      } as any,
+      "query": "*:*",
+      "filter": `docType: PRODUCT AND productId: ${productId}`
+    }
+  }
+
+  let response = {} as Product | Response
+
+  try {
+    const resp = await api({
+      url: 'solr-query',
+      method: 'post',
+      data: query
+    })
+
+    if (resp?.status == 200 && resp.data?.grouped?.groupId?.groups?.length > 0) {
+      const group = resp.data.grouped.groupId.groups[0]
+      const productDetails = group.doclist.docs[0]
+      const imageUrls = productDetails.additionalImageUrls
+      imageUrls.push(productDetails.mainImageUrl)
```

Comment:
I think we don't need to add the main image to the `imageUrls` as we have the mainImage available in another property.

Reply:
updated